Feature/ccclxxxvi 12

tavor118 opened the pull request:
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=993)


## Code reviewers

- [ ] @Andriy97091
- [ ] @irynazaiets
- [ ] @jelysaw
- [ ] @MaksymGromyk
- [ ] @shado1997
- [ ] @tavor118
- [ ] @VolodymyrBodnar
- [ ] @wm139


### Second Level Review

- [ ] @lhalam
- [ ] @alutsysh

## Summary of issue

Added statistic.

## Summary of change

Added views and templates for statistic.
Added functions to work with datetime.
Added AJAX for POST requests from statistic page.

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

tavor118 committed:
Merge branch 'dev' into feature/CCCLXXXVI-18

tavor118 committed:
Merge branch 'dev' into feature/CCCLXXXVI-20

tavor118 committed:
Added statistic.

Added class for statistic.
Added functions for working with datetime.
Added function for working with currency rates.
Added view, template for sraristic.
Added AJAX for POST requests.

tavor118 committed:
Merge branch 'dev' into feature/CCCLXXXVI-12

alutsysh commented on moneta/src/python/core/utils.py:
```diff
@@ -1,12 +1,24 @@
 from settings.settings import DATABASES  # pylint:disable = no-name-in-module, import-error
+from src.python.core.db.redis_worker import RedisWorker
+
+CURRENCY_LIST = ['UAH', 'USD', 'EUR', 'GBP']
```
do you need this?
You should use info from DB

alutsysh commented on moneta/src/python/db/stat_inform.py:
```diff
@@ -0,0 +1,264 @@
+        :params user_id: id of logged user
+        :return: list of available years
+        """
+        sql = f"""
```
do you really need sub select?

alutsysh commented on moneta/src/python/db/stat_inform.py:
```diff
@@ -0,0 +1,264 @@
+            ;
+            """
+        args = (user_id, start, end)
+        query = Statistic._make_select(sql, args)
```
it is not a query. rename

MaksymGromyk committed:
DATABASE STRUCTURE MODIFICATIONS

ADDED Foreign key to user_settings,
CHANGED name from user to user_settings,
ADDED table Currencies INSERT INTO currencies (cur),
ADDED create_time to transactions, changed amount and cur_coef to amount_from and amount_to,
MOVED is_owner from expend/current to user_expend/user_current,
DROPED table user_income,
ADDED owner_id to current, expend, income.

tavor118 committed:
Changes after review.

tavor118 committed:
Added class for interacting with currencies.

tavor118 committed:
Merge branch 'dev' into feature/CCCLXXXVI-12
